sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/RestrictDeserializedTypes.cs

```diff
@@ -103,11 +103,50 @@ public override ProgramState ObjectCreated(ProgramState programState, SymbolicVa
                     {
                         programState = programState.SetConstraint(symbolicValue, SerializationConstraint.Unsafe);
                     }
+
+                    if (IsLosFormatter(typeSymbol) &&
+                        !HasSafeConstructorInitialization(objectCreation, programState))
+                    {
+                        // For LosFormatter the rule is raised directly on the constructor.
+                        addNode(objectCreation);
+                    }
                 }
 
                 return base.ObjectCreated(programState, symbolicValue, instruction);
             }
 
+            private bool HasSafeConstructorInitialization(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                // The constructor is safe only if it has 2 arguments and the first argument value is true.
+                if (objectCreation.ArgumentList.Arguments.Count != 2)
+                {
+                    return false;
+                }
+
+                var firstArgument = objectCreation.ArgumentList.Arguments[0].Expression;
+                if (firstArgument.IsKind(SyntaxKind.FalseLiteralExpression))
+                {
+                    return false;
+                }
+
+                if (firstArgument.IsKind(SyntaxKind.TrueLiteralExpression))
+                {
+                    return true;
+                }
+
+                var symbol = semanticModel.GetSymbolInfo(firstArgument).Symbol;
+                if (symbol == null)
+                {
+                    // In this case we cannot determine if the first parameter is true or false so we
+                    // assume it's true to avoid FPs.
+                    return true;
+                }
```

Comment:
Same here: for readability, I think we could merge these 2 early return together too

Reply:
As discussed offline, the fact that `symbol` is used bellow it makes it harder to join the conditions in a more readable way.